src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -432,26 +452,90 @@ record Element(Label target, Label next, Object caseLabel) {}
                         Label next = element.next();
                         cb.labelBinding(element.target());
                         if (element.caseLabel() instanceof Class<?> classLabel) {
-                            Optional<ClassDesc> classLabelConstableOpt = classLabel.describeConstable();
-                            if (classLabelConstableOpt.isPresent()) {
-                                cb.aload(0);
-                                cb.instanceof_(classLabelConstableOpt.orElseThrow());
-                                cb.ifeq(next);
+                            if (unconditionalExactnessCheck(selectorType, classLabel)) {
+                                //nothing - unconditionally use this case
+                            } else if (classLabel.isPrimitive()) {
+                                if (!selectorType.isPrimitive() && !Wrapper.isWrapperNumericOrBooleanType(selectorType)) {
+                                    // Object o = ...
+                                    // o instanceof Wrapped(float)
+                                    cb.aload(0);
```

Comment:
probably just a matter of style so up to you, but I don't like the mixing of low level code generation here with higher level logic. I would prefer to see the code generation be extracted if possible to helper methods and or a separate helper class.

Reply:
see my other comment above that I think supersedes this one

Reply:
Done in https://github.com/openjdk/jdk/pull/15638/commits/13bfc43654f23f842ac71c1971de32da209d64d3